### PR TITLE
[10.0] fix and improve account_analytic_no_lines

### PR DESCRIPTION
**Pull request by sbidoul**
- the module was not actually doing what it was advertised to do :disappointed: (it was hiding analytic entries but was still generating them in some cases), moreover the name of the group was wrong
- improve it to display debit/credit/balance computed from GL entries

### CODE REVIEW

**Comment by sbidoul** [comment]
@gurneyalex do you know why the In progress label was added automatically (and what it means)?

**Comment by sbidoul** [comment]
@ThomasBinsfeld @alexis-via

File: account_analytic_no_lines/models/account_analytic_account.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+
+class AccountAnalyticAccount(models.Model):
+
+    _inherit = 'account.analytic.account'
+
+    @api.multi
+    def _compute_gl_debit_credit_balance(self):
+        aml_obj = self.env['account.move.line']
+        domain = [('analytic_account_id', 'in', self.mapped('id'))]
```
Comment (ThomasBinsfeld) [inline account_analytic_no_lines/models/account_analytic_account.py]: @sbidoul `self.mapped('id')` can be written `self.ids`, no ?

File: account_analytic_no_lines/models/account_analytic_account.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        if self._context.get('from_date', False):
```
Comment (ThomasBinsfeld) [inline account_analytic_no_lines/models/account_analytic_account.py]: @sbidoul `False` is not required.

File: account_analytic_no_lines/tests/test_account_analytic_no_lines.py [hunk 3]
```diff
@@ -135,3 +146,12 @@ def test_finalize_invoice_move_lines_2(self):
...
         self.assertFalse(analytic_lines)
+
+    def test_gl_amounts(self):
+        self.invoice.action_move_create()
+        self.assertEquals(self.aa_1.gl_debit, 150)
+        self.assertEquals(self.aa_1.gl_credit, 0)
+        self.assertEquals(self.aa_1.gl_balance, -150)
+        self.assertEquals(self.aa_2.gl_debit, 0)
+        self.assertEquals(self.aa_2.gl_credit, 100)
+        self.assertEquals(self.aa_2.gl_balance, 100)
```
Comment (ThomasBinsfeld) [inline account_analytic_no_lines/tests/test_account_analytic_no_lines.py]: @sbidoul `assertEquals` is deprecated, use `assertEqual`

**Review by alexis-via — APPROVED** [review]

**Review by ThomasBinsfeld — APPROVED** [review]

**Comment by rafaelbn** [comment]
Hello @sbidoul ,

I ask here because it look you fixed with this PR the issue I have everytime I test a module in /account-analytic

The security group says:

> Hide Analytic Entries




But when the group is disabled it hide analytic entries and when is enabled it show analytic entries.

Is it wrong? Shouldn't it do the opposite? 😕 

Please, could you clarift me this? Thank you a lot!

**Comment by sbidoul** [comment]
@rafaelbn that is strange it should display View Analytic Entries: https://github.com/OCA/account-analytic/pull/104/files#diff-8ee4653533629e778a82cec07e968699

Do you have the last version?

**Comment by rafaelbn** [comment]
Hi @sbidoul , I tested in runbot:

http://3307230-127-ace56c.runbot1.odoo-community.org/web?debug#id=1&view_type=form&model=res.users&menu_id=65&action=71

Testing this PR: https://github.com/OCA/account-analytic/pull/127

Surely the PR is not up-to-date

**Pull request merged**